app/controllers/advertisements_controller.rb

```diff
@@ -1,21 +1,21 @@
 class AdvertisementsController < ApplicationController
   def index
     @advertisements = Advertisement.order(created_at: :desc)
-  end
-
-  def show
-    @advertisement = Advertisement.find(params[:id])
+    authorize! @advertise
```

Comment:
こちらでも動くのかな？という気がしたんですが動かないですかね…？
```suggestion
    authorize!
```